File: 2 semestre/Técnica de Desenvolvimento de Algoritmos/Listas de Exercicios/Lista 3/ex6.c
#include <stdio.h>

main()
{
    printf("Numero Perfeito\n\n");

    //vars
    int num, divisor, soma=0;

    //Entrada
    printf("Digite um numero inteiro positivo: ");
    scanf("%d", &num);

    //Processamento
    for(divisor=1; divisor<num; divisor++)
    {
        if( num%divisor == 0 )
        {
            soma += divisor;
        }

    }

    //Saída
    if( num == soma )
        printf("%i eh perfeito", num);
    else
        printf("%d NAO eh perfeito", num);


}

File: 2 semestre/Laboratório de Desenvolvimento de Algoritmos/Lista de Exercícios 3/src/Ex6.java
//Importando Scanner
import java.util.Scanner;

public class Ex6{
	
	public static void main(String[] args){
		
		//Instanciando Scanner
		Scanner input = new Scanner(System.in);
		
		System.out.println("Número Perfeito\n\n");
		
		//vars
		int num, divisor, soma=0;
		
		//Entrada
		System.out.print("Insira um número inteiro: ");
		num = input.nextInt();
		
		//Processamento
		for( divisor=1; divisor<num; divisor++ )
		{
			if( num%divisor == 0 )
			{
				soma += divisor;
			}
			
		}
		
		//Saída
		if( soma == num )
			System.out.println(num + " é perfeito!");
		else
			System.out.println(num + " NÃO é perfeito!");
		
	}
}